python-semantic-release/python-semantic-release
# feat(parsers): enable parsers to identify linked issues on a commit

## Purpose


- Add functionality to built-in parsers to parse out issue references from commit messages

## Rationale


This adds essential functionality for PSR to extract issue closure references defined in commits and include them in the Parsed Commit object. This then allows consumers custom changelog to display issue information as desired.

I did not add a default method to display closed issues in the default changelog as I have not decided on a standard format that would be accepted by the community. Only time will tell. The custom changelog for PSR will have a different format for the changelog than the release notes, so we will see if people request a specific default to be added.

## How did you test?


Added a significant number of parameterized values to attempt to test all boundaries of different variations of git footer usage of closure statements. The closure statements are all defined by the different VCS hosting platforms.

## How to Verify


1. Checkout this pr
2. Reset by 7 commits to remove the fix
3. Run pytest to see the failures on issues that are not identified
4. Fast forward merge to the latest commit of this pr
5. Run pytest again to see it is fixed.
